x-pack/plugins/fleet/public/plugin.ts

```diff
@@ -253,25 +255,30 @@ export class FleetPlugin implements Plugin<FleetSetup, FleetStart, FleetSetupDep
     return {
       // Temporarily rely on superuser check to calculate authz. Once Kibana RBAC is in place for Fleet this should
       // switch to a sync calculation based on `core.application.capabilites` properties.
-      authz: getPermissions()
+      authz: Promise.all([getPermissions(), deps.security?.authc.getCurrentUser()])
         .catch((e) => {
           // eslint-disable-next-line no-console
           console.warn(`Could not load Fleet permissions due to error: ${e}`);
-          return { success: false };
+          return [{ success: false }, undefined] as [
+            { success: boolean },
+            AuthenticatedUser | undefined
+          ];
         })
-        .then((permissionsResponse) => {
-          if (permissionsResponse.success) {
+        .then(([permissionsResponse, user]) => {
+          if (permissionsResponse?.success) {
             // If superuser, give access to everything
             return calculateAuthz({
               fleet: { all: true, setup: true },
               integrations: { all: true, read: true },
+              isSuperuser: user ? user.roles.includes('superuser') : false,
```

Comment:
Is the roles check here necessary? Won't `permissionsResponse.success` only be true if the user is superuser? Guess it can't hurt to have an explicit check for later. I think in either case, we'll be able to pass `false` in here later once the only actions that require superuser are not used by UI code (like `uploadPackage`).

Reply:
Yes you are correct for now this will always be true. I can remove that code 👍